### ISSUE TITLE
[un-sdg] svg loading component for UN sustinable development goals

### ISSUE THREAD

**Issue by btopro** [haxtheweb/issues#2147]
https://unfoundation.org/what-we-do/issues/sustainable-development-goals/ lays out 17 goals for building a more sustainable world. While not directly affecting HAX, we do align heavily with multiple goals here and it's entirely possible that education based material would want to demark it's alignment with these goals

That's why we should make it easier to show that these goals are being sought. Right now someone would have to upload an image each time. This is cumbersome but also bc there is text in the image it would have to have accessibility enhancements applied each time. The goals will get out of alignment potentially across solutions. SVG + web components would provide a simple solution to this problem

https://commons.wikimedia.org/wiki/File:Sustainable_Development_Goals.svg has SVGs for all of the goals. Ideally we'd have a very simple API that allows these goals to be presented without constantly having to remember the goals, images, or file names. ALT text could automatically be generated per goal images too for enhanced accessibility considerations baked in.

# Requirements
- property that allows for values of 1-17, "all" or "circle" called `goal`
- 1-17 would just present the single goal by referencing the file in question. Probably a large `switch` statement or method of referencing a single file and then changing the file name such as `lib/svgs/goal-${x}.svg` where x is the goal in question
- ability to override the default `alt` text via a `label` property, however with no `alt` text defined it should always default to the goal name / statement as to what the image is automatically
- Ability to size the image however you want though the `un-sdg` element should supply default width that is reasonable sized (say 254px / 254px as that is the standard set on the website communicating about them)
- it should be stateful so that if the goal changes, the image changes. If the `label` text is not defined then it should default to whatever the associated new `alt` text is relative to the goal (so it is updating based on changes to the goal, the goal is the source of truth)
- image should have lazy `loading` attributes and low `fetchpriority` by default but support properties to changes these defaults

## Additional requirements
- size should be able to be set via height/width if not on `:host` then via css variable that is passed down
- There is a color associated with each of the goals 1-17
- Create a CSS variable for each color code like `--un-sdg-goal-10: #e01a83;`
- Add a `Boolean` called `color-only` which takes the goal and renders a square but without the text
- This could be useful for presenting a square that is the color
- code up on github
- vercel  demo live rebuilding to see output

# Recommended steps to complete this
- need all the images to be in a folder in your element so that you can reference them
- SVGs can either be loaded and inlined OR pointed to as the file in question
- either approach will require the use of `new URL('./lib/svgs/goal-1.svg', import.meta.url).href` as the way of establishing the path to where the files live. This can be dynamically evaluated.
- There should be no calls to `shadowRoot` for this element and it shouldn't have a slot as these provide no value
- You should use `updated()` lifecycle hook in your element to react to changes to `goal` and automatically assign the associated image and alt text relative to that change.
- The SVGs themselves should give you the hex color code for each goal. Opening these files in Chrome you should be able to right click and inspect to obtain the exact color code

# Demo requirements
- Your demo should have all 17 goals + the circle and the "all"
- It should also have an element demonstrating the `color-only` option
- Your demo should have a button that changes the goal of the first element to be a random number 1-17 to illustrate statefulness

We will work on this over several class periods as a mix of together + critique for feedback + tutorials based on questions that come up along the way.

# Bonus
The top implementations, meaning the ones that would require less than an hour of work by our team to put into production, will receive +2 bonus points (so 150% on this assignment). This means it is meeting all of the requirements above and has implemented recommendations and feedback in class. At most only **3 pods** will be awarded this distinction for most perfect implementation as judged by our LAs and professor.

**Comment by btopro** [comment]
![image](https://github.com/user-attachments/assets/ce601f80-fe91-4c9d-a280-af33c2cac793)

**Comment by btopro** [comment]
https://www.npmjs.com/package/@haxtheweb/un-sdg
